awx/network_ui/consumers.py

```diff
@@ -0,0 +1,277 @@
+# Copyright (c) 2017 Red Hat, Inc
+# In consumers.py
+from channels import Group, Channel
+from channels.sessions import channel_session
+from awx.network_ui.models import Topology, Device, Link, Client, Interface
+from awx.network_ui.models import TopologyInventory
+import urlparse
+from django.db.models import Q
+from collections import defaultdict
+import logging
+
+
+from awx.network_ui.utils import transform_dict
+from pprint import pformat
+
+import json
+# Connected to websocket.connect
+
+HISTORY_MESSAGE_IGNORE_TYPES = ['DeviceSelected',
```

Comment:
I would find this easier to read as less lines.

Reply:
I'll remove this as it is no longer needed.  The undo/redo feature was removed for this release.